platform/security/certificate/security-certificate-generator/src/test/java/org/codice/ddf/security/certificate/generator/CertificateCommandTest.java

```diff
@@ -0,0 +1,228 @@
+                is(false));
+
+        assertThat(CertificateCommand.configureDemoCertWithDN(new String[] {"cn=John Whorfin",
+                        "o=Yoyodyne", "l=San Narciso", "st=California", "c=US"}, null),
+                equalTo("CN=John Whorfin,O=Yoyodyne,L=San Narciso,ST=California,C=US"));
+
+        validateKeyStore("john whorfin", false);
+    }
+
+    @Test
+    public void testConfigureDemoCertWithDNAndSANs() throws Exception {
+        final KeyStoreFile ksf = CertificateCommand.getKeyStoreFile();
+
+        assertThat(ksf.aliases()
+                .size(), is(2));
+        assertThat(ksf.isKey("CN=John Whorfin,O=Yoyodyne,L=San Narciso,ST=California,C=US"),
+                is(false));
+
+        assertThat(CertificateCommand.configureDemoCertWithDN(new String[] {"cn=John Whorfin",
+                        "o=Yoyodyne", "l=San Narciso", "st=California", "c=US"},
+                CertificateCommandTest.SANs),
+                equalTo("CN=John Whorfin,O=Yoyodyne,L=San Narciso,ST=California,C=US"));
+
+        validateKeyStore("john whorfin", true);
+    }
+
+    @Test
+    public void testMainWithCNAndWithoutSAN() throws Exception {
+        CertificateCommand.main(new String[] {"-cn", "my-fqdn"});
+
+        validateKeyStore("my-fqdn", false);
+    }
+
+    @Test
+    public void testMainWithCNAndSANs() throws Exception {
+        CertificateCommand.main(new String[] {"-cn", "my-fqdn", "-san", CertificateCommandTest.SANs_ARG});
+
+        validateKeyStore("my-fqdn", true);
+    }
+
+    @Test
+    public void testMainWithCNAndSANsReversedArguments() throws Exception {
+        CertificateCommand.main(new String[] {"-san", CertificateCommandTest.SANs_ARG, "-cn", "my-fqdn"});
+
+        validateKeyStore("my-fqdn", true);
+    }
+
+    @Test
+    public void testMainWithDNAndWithoutSAN() throws Exception {
+        CertificateCommand.main(new String[] {"-dn", "CN=John Whorfin,O=Yoyodyne,L=San Narciso,ST=California,C=US"});
+
+        validateKeyStore("john whorfin", false);
+    }
+
+    @Test
+    public void testMainWithDNAndSANs() throws Exception {
+        CertificateCommand.main(new String[] {"-dn", "CN=John Whorfin,O=Yoyodyne,L=San Narciso,ST=California,C=US", "-san", CertificateCommandTest.SANs_ARG});
+
+        validateKeyStore("john whorfin", true);
+    }
+
+    @Test
+    public void testMainWithDNAndSANsReversedArguments() throws Exception {
+        CertificateCommand.main(new String[] {"-san", CertificateCommandTest.SANs_ARG, "-dn", "CN=John Whorfin,O=Yoyodyne,L=San Narciso,ST=California,C=US"});
+
+        validateKeyStore("john whorfin", true);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testMainWithTooFewArguments() throws Exception {
+        CertificateCommand.main(new String[] { "something" });
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testMainWithTooFewArgumentsWithSAN() throws Exception {
```

Comment:
I can add it; it was simply not testing any new paths or conditions in the code so I left it out.